core/src/main/scala/org/bitcoins/core/wallet/fee/FeeUnit.scala

```diff
@@ -25,28 +25,45 @@ case class SatoshisPerByte(currencyUnit: CurrencyUnit) extends BitcoinFeeUnit {
   def toSatPerKb: SatoshisPerKiloByte = {
     SatoshisPerKiloByte(currencyUnit.satoshis * Satoshis(1000))
   }
+
+  override def calc(tx: Transaction): CurrencyUnit =
+    Satoshis(tx.byteSize * toLong)
 }
 
 object SatoshisPerByte {
   def fromLong(sats: Long): SatoshisPerByte = SatoshisPerByte(Satoshis(sats))
 }
 
+/**
+  * KiloBytes here are defined as 1000 bytes.
+  */
 case class SatoshisPerKiloByte(currencyUnit: CurrencyUnit)
     extends BitcoinFeeUnit {
 
-  def toSatPerByte: SatoshisPerByte = {
-    val conversionOpt = (currencyUnit.toBigDecimal * 0.001).toBigIntExact
+  lazy val toSatPerByteExact: SatoshisPerByte = {
+    val conversionOpt = (currencyUnit.toBigDecimal / 1000.0).toBigIntExact
     conversionOpt match {
       case Some(conversion) =>
         val sat = Satoshis(conversion)
         SatoshisPerByte(sat)
 
       case None =>
         throw new RuntimeException(
-          s"Failed to convert sat/kb -> sat/byte for ${currencyUnit}")
+          s"Failed to convert exactly sat/kb -> sat/byte for ${currencyUnit}")
```

Comment:
nit: I'd prefer `s"Failed to convert sat/kb -> sat/byte (loss of precision)  for $currencyUnit"`